client/executor/src/wasm_runtime.rs

```diff
@@ -373,13 +364,13 @@ fn decode_runtime_apis(apis: &[u8]) -> Result<Vec<([u8; 8], u32)>, WasmError> {
 /// sections, `Err` will be returned.
 pub fn read_embedded_version(blob: &RuntimeBlob) -> Result<Option<RuntimeVersion>, WasmError> {
 	if let Some(mut version_section) = blob.custom_section_contents("runtime_version") {
-		// We do not use `decode_version` here because the runtime_version section is not supposed
+		// We do not use `RuntimeVersion::decode` here because the runtime_version section is not supposed
 		// to ever contain a legacy version. Apart from that `decode_version` relies on presence
 		// of a special API in the `apis` field to treat the input as a non-legacy version. However
 		// the structure found in the `runtime_version` always contain an empty `apis` field.
 		// Therefore the version read will be mistakenly treated as an legacy one.
-		let mut decoded_version = sp_api::RuntimeVersion::decode(&mut version_section)
-			.map_err(|_| WasmError::Instantiation("failed to decode version section".into()))?;
+		let mut decoded_version = sp_version::RuntimeVersionLatest::decode(&mut version_section)
+			.map_err(|_| WasmError::Instantiation("failed to decode version section".into()))?.0;
```

Comment:
@bkchr , here is the case where I need to use RuntiemVersionLatest. If I don't, then the state version and transaction version are incorrect because in this case the apis versions are not included in the RuntimeVersion.

Reply:
Below we are decoding the runtime apis. We could do this first and then add some function "decode_with_version_hint".

And also add some static function to `RuntimeVersion` for getting the version from the given runtime apis.

Reply:
Yes, so decode will be using the decoding without the apis check, and we have specific code for decoding with compatibility.
Looks safer to me.
But in this case am I correct if I say that we need to restore the core runtime :
```
// Returns the version of the runtime.
		#[changed_in(4)]
		fn version() -> RuntimeVersionV1;
		/// Returns the version of the runtime.
		#[changed_in(3)]
		fn version() -> RuntimeVersionV0;
```

Reply:
No. This code here is not called by the runtime api.

Reply:
(It is just implicitly called, however this function is for fetching the version from a wasm section and the runtime api is about calling into the runtime, aka something different)